src/Cli/dotnet/Program.cs

```diff
@@ -31,10 +31,13 @@ public static int Main(string[] args)
     {
         using AutomaticEncodingRestorer _ = new();
 
-        // Setting output encoding is not available on those platforms
-        if (UILanguageOverride.OperatingSystemSupportsUtf8())
+        if (Environment.GetEnvironmentVariable("CONSOLE_USE_DEFAULT_ENCODING") != "1")
```

Comment:
Any env variables that are important to the dotnet CLI need a `DOTNET_CLI_` prefix, and then we'd need to 'transform' them to the MSBuild-friendly version before we call out to MSBuild in the MSBuildforwardingApp.cs logic.